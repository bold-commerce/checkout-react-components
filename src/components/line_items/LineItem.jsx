import React from 'react';
import PropTypes from 'prop-types';
import Product from '../product/Product';

const LineItem = ({
  title,
  image,
  quantity,
  price,
  totalPrice,
  lineItemKey,
  description,
  onQuantityChange,
  onRemove,
  readOnly = false,
}) => (
  <div className="SummaryBlock CartItem" key={lineItemKey}>
    <Product
      title={title}
      image={image}
      quantity={quantity}
      itemPrice={price}
      totalPrice={totalPrice}
      lineItemKey={lineItemKey}
      description={description}
      onQuantityChange={onQuantityChange}
      onRemove={onRemove}
      readOnly={readOnly}
    />
  </div>
);

LineItem.propTypes = {
  title: PropTypes.string,
  image: PropTypes.string,
  quantity: PropTypes.number,
  price: PropTypes.number,
  totalPrice: PropTypes.number,
  lineItemKey: PropTypes.string,
  description: PropTypes.string,
  onQuantityChange: PropTypes.func,
  onRemove: PropTypes.func,
  readOnly: PropTypes.bool,
};

export default LineItem;
