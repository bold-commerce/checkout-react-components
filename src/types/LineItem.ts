export interface LineItem {
  platform_id: string,
  sku: string,
  quantity: number,
  line_item_key: string
}