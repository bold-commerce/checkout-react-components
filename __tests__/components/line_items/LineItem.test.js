import React from 'react';
import { render } from '@testing-library/react';
import LineItem from '../../../src/components/line_items/LineItem';

describe('LineItem', () => {
  test('renders LineItem component with editable quantity', () => {
    const { asFragment } = render(
      <LineItem
        title="Product A"
        image="https://via.placeholder.com/150"
        quantity={2}
        price={9999}
        totalPrice={19998}
        lineItemKey="abcd"
        onRemove={() => {}}
        onQuantityChange={() => {}}
        description="Test description right here"
        readOnly={false}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders read only LineItem component', () => {
    const { asFragment } = render(
      <LineItem
        title="Product A"
        image="https://via.placeholder.com/150"
        quantity={2}
        price={9999}
        totalPrice={19998}
        lineItemKey="abcd"
        onRemove={() => {}}
        onQuantityChange={() => {}}
        description="Test description right here"
        readOnly
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
