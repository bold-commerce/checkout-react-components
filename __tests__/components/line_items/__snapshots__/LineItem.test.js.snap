// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineItem renders LineItem component with editable quantity 1`] = `
<DocumentFragment>
  <div
    class="SummaryBlock CartItem"
  >
    <div
      class="CartItem__ImageDescriptionWrapper"
    >
      <div
        class="CartItem__ProductImageWrapper"
      >
        <div
          class="stx-image"
        >
          <div
            class="stx-image--content"
          >
            <div
              class="stx-image--holder"
            >
              <img
                alt="Product A"
                class="stx-image--image"
                src="https://via.placeholder.com/150"
                title="Product A"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="CartItem__ProductDetails"
      >
        <div
          class="stx-details stx-details--left"
        >
          <div
            class="stx-details__title"
          >
            Product A
          </div>
          <div
            class="stx-details__description"
          >
            Test description right here
          </div>
        </div>
        <button
          aria-busy="false"
          aria-disabled="false"
          class="stx-button stx-button--secondary"
        >
          <span
            class="stx-button__text"
          >
            Remove
          </span>
          <span
            class="stx-loading-spinner stx-button__spinner"
          />
        </button>
      </div>
    </div>
    <div
      class="CartItem__QuantityPriceWrapper"
    >
      <div
        class="CartItem__ProductQuantityWrapper"
      >
        <div
          class="ProductQuantity__Decrement"
        >
          <button
            aria-busy="false"
            aria-disabled="false"
            class="stx-button"
          >
            <span
              class="stx-button__text"
            >
              -
            </span>
            <span
              class="stx-loading-spinner stx-button__spinner"
            />
          </button>
        </div>
        <div
          class="ProductQuantity"
        >
          <div
            class="stx-input"
          >
            <input
              aria-readonly="false"
              class="stx-input__input-element"
              type="number"
              value="2"
            />
          </div>
        </div>
        <div
          class="ProductQuantity__Increment"
        >
          <button
            aria-busy="false"
            aria-disabled="false"
            class="stx-button"
          >
            <span
              class="stx-button__text"
            >
              +
            </span>
            <span
              class="stx-loading-spinner stx-button__spinner"
            />
          </button>
        </div>
      </div>
      <div
        class="CartItem__ProductPrice"
      >
        <div
          class="stx-price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            CA$199.98
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LineItem renders read only LineItem component 1`] = `
<DocumentFragment>
  <div
    class="SummaryBlock CartItem"
  >
    <div
      class="CartItem__ImageDescriptionWrapper"
    >
      <div
        class="CartItem__ProductImageWrapper"
      >
        <div
          class="stx-image"
        >
          <div
            class="stx-image--content"
          >
            <div
              class="stx-image--holder"
            >
              <img
                alt="Product A"
                class="stx-image--image"
                src="https://via.placeholder.com/150"
                title="Product A"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="CartItem__ProductDetails"
      >
        <div
          class="stx-details stx-details--left"
        >
          <div
            class="stx-details__title"
          >
            Product A
          </div>
          <div
            class="stx-details__description"
          >
            Test description right here
          </div>
        </div>
      </div>
    </div>
    <div
      class="CartItem__QuantityPriceWrapper"
    >
      <div
        class="CartItem__ProductQuantityWrapper"
      >
        <div
          class="ProductQuantity__Decrement"
        />
        <div
          class="ProductQuantity ProductQuantity--read-only"
        >
          <div
            class="stx-input stx-input--is-read-only"
          >
            <input
              aria-readonly="true"
              class="stx-input__input-element"
              readonly=""
              type="number"
              value="2"
            />
          </div>
        </div>
        <div
          class="ProductQuantity__Increment"
        />
      </div>
      <div
        class="CartItem__ProductPrice"
      >
        <div
          class="stx-price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            CA$199.98
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
